runuser: add page (#3283)
tldr-pages/tldr: pages/linux/runuser.md

# runuser

> > Run commands as a specific user and group without asking for password (needs root privileges).

Reviewer: If it needs root privileges, perhaps we could prefix the commands themselves with `sudo` instead of mentioning it in the description?

This is what we've done in other pages.

Reviewer 2: Hmm, this is intended to be literally run as root though, since it does not ask for password. Running it with sudo defeats the purpose of the command, for that you have `su`.

Reviewer: Ah, I see! In that case this might be fine then.

> > Run commands as a specific user and group IDs. Does not ask for password. Only useful when run as the root user.

Reviewer 2: ```suggestion
> Run commands as a specific user and group without asking for password (needs root privileges).
```

Now that I take a closer look at it, I think this can be simplified in a shorter sentence. What do you think?

Author: Agree, thanks for reviewing.

- Run command as a different user:

`runuser {{user}} -c '{{command}}'`

- Run command as a different user and group:

`runuser {{user}} -g {{group}} -c '{{command}}'`

- Start a login shell as a specific user:

`runuser {{user}} -l`

> - Specify a [s]hell for running instead of default shell (also works for login):

Reviewer 2: ```suggestion
- Specify a shell for running instead of the default shell (also works for login):
```
Sorry missed the article there, also my bad didn't mean to use `[s]`, copy-paste error. :sweat_smile:

Author: oh, I think it's format style from our guideline 🙏 ...

Author: Btw, it's OK

> - Specific a shell for running or login instead of default shell:

Reviewer 2: ```suggestion
- Specify a [s]hell for running instead of default shell (also works for login):
```
Typo. Also, since the command we show only does one of the two things, let's reword to say it *can* also be used for login.

Author: Agree, thanks for reviewing.

> `runuser {{user}} -l -s /bin/sh`

Reviewer 2: ```suggestion
`runuser {{user}} -s {{/bin/sh}}`
```
Wrap shell path in tokens and remove `-l` as per the previous comment.

Author: Drop `-l` is OK, no need to have this option, thanks for reviewing.

> - Preserve the entire environment of root (if running by root). Ignore when having option `--login`:

Reviewer 2: ```suggestion
- Preserve the entire environment of root (only if `--login` is not specified):
```
Simplified a bit, since the command can basically only be run by root.

Author: Great. Thanks for reviewing.

# runuser

> > Run commands as a specific user and group IDs.

Reviewer: Perhaps this will read better?
```suggestion
> Run commands as specific users and group IDs.
```

Author: `runuser` only run as a specific user at one time, not users, disagree.

> > Do not ask for password. Only useful when run as root user.

Reviewer: Perhaps this will read better?
```suggestion
> Does not ask for password. Only useful when run as the root user.
```

Author: Agree, thanks for reviewing

> - Run 'command' under substitute 'user':

Reviewer: Perhaps this will read better?
```suggestion
- Run a command under a substitute user:
```

Author: `runuser` can run many commands or even some scripts, not only one command. Disagree.

`runuser {{user}} -c '{{command}}'`

> - Run 'command' under substitute 'user' and 'group':

Reviewer: Perhaps this will read better?
```suggestion
- Run a command under a substitute user and group:
```

Author: Same as above, `runuser` can run many commands or even some scripts, not only one command. Disagree.

`runuser {{user}} -g {{group}} -c '{{command}}'`

- Start a login shell as a specific user:

`runuser {{user}} -l`

> - Specific a 'shell' for running or login instead of default shell:

Reviewer: Perhaps this will read better?
```suggestion
- Start a specific shell instead of the default shell:
```

Author: `for running or login` is describing exactly the feature of `runuser`. Disagree.

`runuser {{user}} -l -s /bin/sh`

> - Preserve the entire environment of root (if running by root). Ignore when having option `--login`:

Reviewer: Perhaps this will read better?
```suggestion
- Inherit the current environment (ignored when using the `--login` option):
```

Author: The original sentence is describing exactly the concept of **preserve** env from `runuser` or even `su` philosophy, we cannot use **inherit** in this case. Disagree.
